dev/archery/archery/lang/cpp.py

```diff
@@ -56,7 +56,7 @@ def __init__(self,
                  with_ipc=True, with_json=None, with_jni=None,
                  with_mimalloc=None, with_jemalloc=None,
                  with_parquet=None, with_plasma=None, with_python=True,
-                 with_r=None, with_s3=None,
+                 with_r=None, with_gcs=None, with_s3=None,
```

Comment:
Please, let's keep this "components" list alphabetically ordered.
Also, while we're at it, can you make all those constructor arguments keyword-only?
```python

class CppConfiguration:
    def __init__(self, *,
                 # toolchain
                 cc=None, cxx=None, cxx_flags=None,
```

Reply:
I agree, since this could have a wide blast radius I filed: https://issues.apache.org/jira/browse/ARROW-16227 to do this separately.